### DISCUSSION TITLE
How to output the thermal stress?

### DISCUSSION THREAD

**Discussion by RECHOA** [idaholab/moose#22404] [Q&A Modules: Solid mechanics] [answered]
Hi all,

Is there any quantity could represent the `thermal stress` (stress induced by the temperature changes and  the inhomogeneous distribution of temperature ) and output it ?

Thanks in advance.

**Comment by ngrilli** [comment, marked as answer]
@RECHOA  there is an example of adding thermal eigenstrains and visualizing them using AuxVariables here:

https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/test/tests/thermal_expansion/constant_expansion_stress_free_temp.i

Nicolò

**Reply by RECHOA** [reply]
Sorry I'm concerned about the thermal stress rather than the thermal strian. Maybe you means using the `σ=E·ε` to obtain it, where σ is stress, E is the Young's modulus, ε is the thermal strain. 
But it only reflect the thermal stress induced by the temperature changes, without the inhomogeneous distribution of temperature.

**Reply by hugary1995** [reply]
@RECHOA could you write down the expression for the thermal stress you are referring to?

**Reply by RECHOA** [reply]
I'm struggling to find such a expression for the thermal stress, so i turned to the experienced researchers here to help me find a expression or  a quantity of **`stress`** could reflect the temperature influence.
